### PR TITLE
SPUI-196 > Crypto withdrawal flow

### PR DESCRIPTION
**Pull request by eli1stark**

### CODE REVIEW

**Review by VovaMedvediev — CHANGES_REQUESTED** [review]

File: lib/app/shared/features/currency_withdraw/helper/user_will_receive.dart [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import '../../../models/currency_model.dart';
+
+/// Calculates amount that user will receive after
+/// subtraction of the currency fee
+String userWillreceive(String amount, CurrencyModel currency) {
+  final value = double.parse(amount);
+  final fee = currency.fees.withdrawalFee?.size ?? 0;
+
+  return '${value - fee} ${currency.symbol}';
```
[the remark this answers is not in the document]
Comment (eli1stark) [reply lib/app/shared/features/currency_withdraw/helper/user_will_receive.dart:9]: done

File: lib/app/shared/features/currency_withdraw/notifier/withdrawal_amount_notifier/withdrawal_amount_notifier.dart [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      tag: address.tag,
+      address: address.address,
+    );
+  }
+
+  final Reader read;
+  final CurrencyModel currency;
+
+  static final _logger = Logger('WithdrawalAmountNotifier');
+
```
[the remark this answers is not in the document]
Comment (eli1stark) [reply lib/app/shared/features/currency_withdraw/notifier/withdrawal_amount_notifier/withdrawal_amount_notifier.dart:29]: will discuss in TG

File: lib/app/shared/features/currency_withdraw/notifier/withdrawal_confirm_notifier/withdrawal_confirm_notifier.dart [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  final CurrencyModel currency;
+
+  Timer? _timer;
+  late int retryTime;
+  late BuildContext _context;
+  late String _operationId;
+
+  static final _logger = Logger('WithdrawalConfirmNotifier');
+
+  Future<void> _withdrawalInfo() async {
```
[the remark this answers is not in the document]
Comment (eli1stark) [reply lib/app/shared/features/currency_withdraw/notifier/withdrawal_confirm_notifier/withdrawal_confirm_notifier.dart:46]: done

File: lib/app/shared/features/currency_withdraw/notifier/withdrawal_confirm_notifier/withdrawal_confirm_notifier.dart [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      _refreshTimer();
+    }
+  }
+
+  void _refreshTimer() {
+    _timer?.cancel();
+    retryTime = _retryTime;
+
+    _timer = Timer.periodic(
+      const Duration(seconds: 1),
```
[the remark this answers is not in the document]
Comment (eli1stark) [reply lib/app/shared/features/currency_withdraw/notifier/withdrawal_confirm_notifier/withdrawal_confirm_notifier.dart:75]: No, _retryTime is the total duration of the timer. `const Duration(seconds: 1)` is the step of the timer.
They just accidentally matched to 1 second.

File: lib/app/shared/features/currency_withdraw/notifier/withdrawal_preview_notifier/withdrawal_preview_notifier.dart [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  }
+
+  final Reader read;
+  final CurrencyModel currency;
+
+  late BuildContext _context;
+
+  static final _logger = Logger('WithdrawalPreviewNotifier');
+
+  void _updateOperationId(String value) {
```
[the remark this answers is not in the document]
Comment (eli1stark) [reply lib/app/shared/features/currency_withdraw/notifier/withdrawal_preview_notifier/withdrawal_preview_notifier.dart:45]: done

File: lib/app/shared/features/currency_withdraw/view/currency_withdraw.dart [hunk 6]
```diff
@@ -4,100 +4,82 @@ import 'package:flutter_screenutil/flutter_screenutil.dart';
...
-          loading: () {},
-        );
-      },
-      child: Scaffold(
-        appBar: AppBar(
-          title: Text(
-            'Withdraw ${widget.currency.description}',
+    return PageFrame(
+      header: 'Withdraw ${currency.description} (${currency.symbol})',
+      onBackButton: () => Navigator.pop(context),
```
[the remark this answers is not in the document]
Comment (eli1stark) [reply lib/app/shared/features/currency_withdraw/view/currency_withdraw.dart:35]: Swipes are not supported at this stage

[the remark this answers is not in the document]
Comment (eli1stark) [reply lib/app/shared/features/currency_withdraw/view/currency_withdraw.dart:35]: Works

**Pull request merged**